packages/flutter/lib/src/widgets/router.dart

```diff
@@ -414,20 +468,30 @@ enum _IntentionToReportRouteInformation {
   ignore,
 }
 
-class _RouterState<T> extends State<Router<T>> {
+class _RouterState<T> extends State<Router<T>> with RestorationMixin {
   Object? _currentRouteInformationParserTransaction;
   Object? _currentRouterDelegateTransaction;
-  late _IntentionToReportRouteInformation _currentIntentionToReport;
+  _IntentionToReportRouteInformation _currentIntentionToReport = _IntentionToReportRouteInformation.none;
+  final _RestorableRouteInformation _routeInformation = _RestorableRouteInformation();
+
+  @override
+  String? get restorationId => widget.restorationId;
 
   @override
   void initState() {
     super.initState();
     widget.routeInformationProvider?.addListener(_handleRouteInformationProviderNotification);
     widget.backButtonDispatcher?.addCallback(_handleBackButtonDispatcherNotification);
     widget.routerDelegate.addListener(_handleRouterDelegateNotification);
-    _currentIntentionToReport = _IntentionToReportRouteInformation.none;
-    if (widget.routeInformationProvider != null) {
-      _processInitialRoute();
+  }
+
+  @override
+  void restoreState(RestorationBucket? oldBucket, bool initialRestore) {
+    registerForRestoration(_routeInformation, 'route');
```

Comment:
Nit: Could this `'route'` string be a constant so that it's easy to track down where it's coming from?

Reply:
These strings typically only appear within the `restoreState` method and only have to be unique here. Therefore, it has been the style in other places of the framework as well to just define them here in-line since they are not shared elsewhere.